custom checkbox style

jancborchardt opened the pull request:
@raghunayyar here are the icons if you want to have a stab at it. :) We should probably use them in 50% or 70% grey to fit better. I can also add them in white because we likely need that for the log in page checkbox.

cc @owncloud/designers

jancborchardt committed:
custom checkbox style

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/38df7b19-651a-4f4e-a578-1810cf8e2145) was created.

karlitschek commented:
let me know if you need help from someone on that. Would be good to have this in today or tomorrow

raghunayyar commented:
I can do this over the weekend. @karlitschek @jancborchardt

karlitschek commented:
@raghunayyar this would be great. Thanks

raghunayyar committed:
Adds background image for checkboxes

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c95df7e7-22c7-4873-afc7-b01fe3d7e24b) was created.

raghunayyar commented:
@jancborchardt @karlitschek there will be some semantics changes (wherever we use check-boxes)for this because I can't assign a background to the check-box but it is assigned to the label along with it. Please check my commit for that matter.

karlitschek commented:
I'm fine with that. @jancborchardt What do you think?

MorrisJobke commented:
Doesn't work on the login page in firefox:

![bildschirmfoto von 2015-09-06 23-11-56](https://cloud.githubusercontent.com/assets/245432/9706652/b9a5f212-54ec-11e5-8452-1039f786053e.png)

MorrisJobke commented:
Also looks weird in generated previews (with white background):

![bildschirmfoto von 2015-09-06 23-12-49](https://cloud.githubusercontent.com/assets/245432/9706656/dada8254-54ec-11e5-95c9-933be849a092.png)

MorrisJobke commented:
Activity settings on personal page:

![bildschirmfoto von 2015-09-06 23-14-14](https://cloud.githubusercontent.com/assets/245432/9706659/08a75734-54ed-11e5-9221-79c761417712.png)

raghunayyar commented:
It will not, that is what I mentioned. We manually style labels and checkboxes on the respective pages. So, we will have to update it there. I am waiting for a heads up for that. It is a bigger change than it looks. Cc @jancborchardt

jancborchardt commented:
@raghunayyar so should we postpone to 9.0?

raghunayyar commented:
@jancborchardt definitely, I will take care of this in core.

karlitschek commented:
@raghunayyar What do you mean? So no way to have this in the next week before feature freeze?

jancborchardt commented:
@raghunayyar do you think you can manage it for next week? Let me know if you need help.

raghunayyar commented:
My problem is, we don't style check-boxes, we style labels in front of them. This change needs to happen everywhere where we use check-boxes in core as well as apps. I will try my best to get this in next week.

karlitschek commented:
Thanks! Sorry to hear that this is more complex then expected

raghunayyar commented:
Also, I need white checkboxes for login cc @jancborchardt

jancborchardt commented:
@raghunayyar it’s fine if for IE8 (and other browsers where it’s difficult to get right) we just show the default checkboxes. But is it possible to fix this for modern browsers like Chrome, Firefox etc? Would be really awesome! :) I’ll add the white checkboxes right now.

jancborchardt committed:
add white versions of custom checkboxes

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/33eb625e-7985-46f8-847d-2c629f35f0c5) was created.

jancborchardt commented:
@raghunayyar added the white versions.

raghunayyar commented:
@jancborchardt awesome, on it.

raghunayyar committed:
Fixes white and black checkboxes in core, files will be next.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d547283f-7602-4e99-bc52-27c6c55d0a2c) was created.

raghunayyar commented:
I am able to fix this everywhere apart from files as there is a thumbnail that overlays the check-box partially. z-index isn't helping either and I don't want to touch JavaScript for that, do you want to give it a shot @MorrisJobke / @PVince81 / @jancborchardt  ? Apart from that, this PR is complete. :)

karlitschek commented:
looks fine on the personal page but broken in the files view. tested with safari
![c](https://cloud.githubusercontent.com/assets/867445/9837410/e095f380-5a3d-11e5-92e4-e4f73850034f.jpg)

Henni commented:
I think it would make sense to bind the `white` class to the checkbox instead of the label.

karlitschek commented:
So can anyone help wo bring this in today before freeze?
@raghunayyar  ?

DeepDiver1975 commented:
> So can anyone help wo bring this in today before freeze?
> @raghunayyar ?

PR is scheduled for 9.0 - do I miss anything? THX

karlitschek commented:
Well this was scheduled for 8.2
It seems that @MorrisJobke decided to move this to 9.0 Can we please stop doing this without a discussion with the involved people? Thanks.
We wanted to have this in 8.2 as described on the top of the thread.

raghunayyar commented:
> I am able to fix this everywhere apart from files as there is a thumbnail that overlays the check-box partially. z-index isn't helping either and I don't want to touch JavaScript for that, do you want to give it a shot @MorrisJobke / @PVince81 / @jancborchardt ? Apart from that, this PR is complete. :)

if anyone is up for a CSS challenge.

karlitschek commented:
@raghunayyar I thought you volunteered for the challenge? ;-)

raghunayyar commented:
One more try for it in 2 hours.

DeepDiver1975 commented:
so 8.2 ???? highly questionable ....

karlitschek commented:
yes. unfortunately. :-( 
Not happy

Henni committed:
use label:before instead of label

Henni committed:
fix filelist checkboxes

Henni commented:
I fixed the essentials. Maybe we can still merge this for 8.2?

karlitschek commented:
@jancborchardt @LukasReschke   what do you think?

jancborchardt commented:
Looks pretty damn awesome @Henni! :) I’m on it to fix the last details (slightly strange look for the checkboxes, will make them less dark and give em a white background)

jancborchardt committed:
fix checkbox images

jancborchardt commented:
Rebased and fixed the images to make the checkboxes look more awesome.

@Henni @raghunayyar @nickvergessen the only issue left is that the checkboxes are not shown in the Activity notification settings, as mentioned above already (see screenshot).

Henni commented:
@jancborchardt should be fixed by https://github.com/owncloud/activity/issues/391

jancborchardt commented:
WOAH, nice! Everything works :) :+1: 

@karlitschek everything done. Now just waiting for the required statuses to pass – before that it is impossible to press the merge button.

raghunayyar commented:
@Henni this is really really awesome. \o/
I think we need rebasing once more.

jancborchardt commented:
@raghunayyar you did a merge instead of the needed rebase. I fixed that. Let’s see what the tests say …

karlitschek commented:
great guys !!!!

jancborchardt commented:
Giant thanks to heroes @raghunayyar & @Henni! :)

blizzz commented:
FYI, That line https://github.com/owncloud/core/pull/18735/files#diff-463c027b5536faafc395f8fd4d09792bR166 causes checkboxes to be invisible, as reported in https://github.com/owncloud/core/issues/18619#issuecomment-142514194 and https://github.com/owncloud/core/issues/19238 for example

PVince81 commented on core/css/styles.css:
```diff
@@ -158,8 +158,42 @@ textarea:hover, textarea:focus, textarea:active {
+	padding:0;
+	height:auto;
+	width:auto;
+	display: none;
```
How about adding an additional CSS class on the input when wanting to use the new style ?
This way the old checkboxes will still look as before instead of disappear. That would be a fallback solution.

@jancborchardt @Henni

Henni replied:
@PVince81 I thought about this, but this leads to additional code in the long-term.
In my opinion the way we have it now is the cleaner solution.

PVince81 replied:
but has the drawback of breaking checkboxes for every app out there, including third party app store apps including the functionality attached to them.

I'd vote for adding an additional class short-term and once all apps have been migrated to the new style we can make it global and remove the need for it.

Henni replied:
@PVince81 how do we get third party apps to migrate to the new style?
It is quite difficult to deprecate css.
BTW most apps and most checkboxes in core worked out of the box because they already had a label next the checkboxes.

PVince81 replied:
The same way as when deprecating JS/PHP functions: write documentation about how to migrate to the new way + email to mailing list.

In the worst case some apps will then look uglier instead of having broken functionality.

PVince81 replied:
Raised https://github.com/owncloud/core/issues/19304 to continue the discussion

raghunayyar replied:
The breaks were bound to happen as I mentioned in the PR itself but I am super glad they are being taken care asap by @Henni. I will also join during the weekend.
